test/e2e/framework/util.go

```diff
@@ -3200,16 +3201,11 @@ func UpdateDaemonSetWithRetries(c clientset.Interface, namespace, name string, a
 // NodeAddresses returns the first address of the given type of each node.
 func NodeAddresses(nodelist *v1.NodeList, addrType v1.NodeAddressType) []string {
 	hosts := []string{}
+	addrTypes := []v1.NodeAddressType{addrType}
 	for _, n := range nodelist.Items {
-		for _, addr := range n.Status.Addresses {
-			// Use the first external IP address we find on the node, and
-			// use at most one per node.
-			// TODO(roberthbailey): Use the "preferred" address for the node, once
-			// such a thing is defined (#2462).
-			if addr.Type == addrType {
-				hosts = append(hosts, addr.Address)
```

Comment:
This lists all that match, the new function lists one per addrtype.  Who determined this was safe?